socorro/processor/cache_manager.py

```diff
@@ -0,0 +1,378 @@
+
+"""
+This defines the DiskCacheManager application. It's designed to run as
+a standalone application separate from the rest of the processor.
+
+It keeps track of files in a directory and evicts files least recently
+used in order to keep the total size under a max number.
+
+It uses inotify to cheaply watch the files.
+
+To run::
+
+    $ /app/bin/run_processor_cache_manager.sh
+
+"""
+
+from collections import OrderedDict
+import datetime
+import logging
+import os
+import pathlib
+import sys
+import traceback
+
+from boltons.dictutils import OneToOne
+from fillmore.libsentry import set_up_sentry
+from fillmore.scrubber import Scrubber, SCRUB_RULES_DEFAULT
+from inotify_simple import INotify, flags
+import markus
+
+from socorro import settings
+from socorro.lib.libdockerflow import get_release_name, get_version_info
+from socorro.lib.liblogging import set_up_logging
+
+
+LOGGER = logging.getLogger(__name__)
+REPOROOT_DIR = str(pathlib.Path(__file__).parent.parent.parent)
+MAX_ERRORS = 10
+METRICS = markus.get_metrics("processor.cache_manager")
+
+
+def count_sentry_scrub_error(msg):
+    metrics = markus.get_metrics("processor")
+    metrics.incr("sentry_scrub_error", value=1, tags=["service:cachemanager"])
+
+
+class LastUpdatedOrderedDict(OrderedDict):
+    """Store items in the order the keys were last added or updated"""
+
+    def __setitem__(self, key, value):
+        """Create or update a key"""
+        super().__setitem__(key, value)
+        self.move_to_end(key, last=True)
+
+    def touch(self, key):
+        """Update last-updated for key"""
+        self.move_to_end(key, last=True)
+
+    def pop_oldest(self):
+        """Pop the oldest item"""
+        return self.popitem(last=False)
+
+
+def handle_exception(exctype, value, tb):
+    LOGGER.error(
+        "unhandled exception. Exiting. "
+        + "".join(traceback.format_exception(exctype, value, tb))
+    )
+
+
+sys.excepthook = handle_exception
+
+
+class DiskCacheManager:
+    def __init__(self):
+        self.basedir = pathlib.Path(__file__).resolve().parent.parent.parent
+        self.logger = logging.getLogger(__name__ + "." + self.__class__.__name__)
+
+        self.cachepath = pathlib.Path(settings.SYMBOLS_CACHE_PATH).resolve()
+        self.max_size = settings.SYMBOLS_CACHE_MAX_SIZE
+
+        # Set up attributes for cache monitoring; these get created in the generator
+        self.lru = LastUpdatedOrderedDict()
+        self.total_size = 0
+        self.watches = OneToOne()
+        self._generator = None
+        self.inotify = None
+        self.watch_flags = (
+            flags.ACCESS
+            | flags.CREATE
+            | flags.DELETE
+            | flags.DELETE_SELF
+            | flags.MODIFY
+            | flags.MOVED_FROM
+            | flags.MOVED_TO
+        )
+
+    def set_up(self):
+        set_up_logging(
+            local_dev_env=settings.LOCAL_DEV_ENV,
+            logging_level=settings.CACHE_MANAGER_LOGGING_LEVEL,
```

Comment:
I have this use a different setting for the logging level because this can be pretty spammy when things are going and I'm pretty sure the typical scenario is one needs one of the services logging at DEBUG and not all of them.